src/primitives/transaction.cpp

```diff
@@ -161,18 +160,26 @@ std::vector<char> loadTransaction(const std::vector<CMFToken> &tokens, std::vect
             seenCoinbaseMessage = true;
             break;
         }
-        case Consensus::TxInScript: {
-            if (inputs.empty()) throw std::runtime_error("TxInScript before TxInPrevHash");
-            if (inputCount == 0) { // copy all of the input tags
+        case Consensus::TxInputStackItem:
+        case Consensus::TxInputStackItemContinued: {
+            if (signatureCount == -1) { // copy all of the input tags
                 CDataStream stream(0, 4);
                 ser_writedata32(stream, nVersion);
                 for (unsigned int i = 0; i < index; ++i) {
                     tokens[i].Serialize(stream, 0, 4);
                 }
                 txData = std::vector<char>(stream.begin(), stream.end());
             }
+            if (signatureCount < 0 || token.tag == Consensus::TxInputStackItem)
+                signatureCount++;
+            if (static_cast<int>(inputs.size()) <= signatureCount)
```

Comment:
I'm not sure why you static cast, but I think I can get your program do do very weird thing with a 4GB+ transaction.